hueplusplus/LinHttpHandler.cpp

```diff
@@ -98,18 +98,18 @@ std::string LinHttpHandler::send(const std::string &msg, const std::string &adr,
           errCode, std::generic_category(),
           "LinHttpHandler: Failed to write message to socket"));
     }
-    if (bytes == 0) {
+    else if (bytes == 0) {
       break;
     }
-    if (bytes) {
+    else {
       sent += bytes;
     }
   } while (sent < total);
 
   // receive the response
   std::string response;
   total = sizeof(response) - 1;
-  int received = 0;
+  decltype(total) received = 0;
```

Comment:
Why not use size_t here?
```suggestion
  size_t received = 0;
```

Reply:
It was 4:30 AM after an all-nighter, I was in an unfamiliar codebase and took the easiest solution to fixing the compile warning blocking my project from deploying before I could go to bed. :)

I'll fix in a bit.